src/main/java/strategy/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package strategy.operation;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+
+    @Override
+    public void handle(FruitTransaction fruitTransaction) {
+        Integer oldSum = Storage.fruits.get(fruitTransaction.getFruit());
+        if (oldSum <= 0 || oldSum < fruitTransaction.getQuantity()) {
```

Comment:
Change the condition to check if oldSum is null, which means the fruit is not present in the Storage. This can throw NPE